src/interface.rs

```diff
@@ -182,6 +190,39 @@ impl<'a> Interface<'a> {
         screen.flush().unwrap();
     }
 
+    fn skim(&mut self) {
+        self.selection = 0;
+        self.matches = self
+            .history
+            .find_matches(&self.input.command, -1);
```

Comment:
How slow is this when you have a large command history?

Reply:
Thanks to an errant config setting I only have a couple thousand lines of real history, but I whipped up a [script](https://gist.github.com/dmfay/32a327869c40ec2673d622957a7b6289) to seed more. Time is me hitting Esc as soon as skim loads.

```
wc -l .histfile
101838 .histfile

time mcfly search --fuzzy
mcfly search --fuzzy  3.36s user 0.17s system 99% cpu 3.543 total
```

After that three-second load (😬), skim's performance is quite acceptable. I can type just faster than it can process 90-100k elements, but it buffers input and responds without stuttering or glitching.

I did notice some drag after _other_ commands as long as I kept the 100k-line history file -- a bit under a second in between the command completing and my prompt printing again. It went away upon replacing my original .histfile so I think that might be the shell, not mcfly.

Reply:
So, I think we need to find something faster. I currently have 50000 commands in my mcfly history and it launches in ~200ms.

Maybe we can only start searching once you've typed a few characters?

Reply:
I was running with debug symbols! :woman_facepalming:

Built for release it's sub-second, in the 850-920ms range, which is hardly instant but still a far cry from three seconds. The version I installed from my package manager loads the same history in 600-650ms.

Reply:
I'd be open to adding it now that I know it's sub-second and since it's optional. Maybe we should have it also be enable-able with an environment variable so that users can add that to their shell config files before sourcing mcfly?

I'd also love to see how your "%s everywhere" solution does, it might be good enough.

Reply:
The "%s everywhere" solution only does anything different starting on the first keystroke, so the initial load time is the same. At 90-100k elements, it's about a quarter-second faster than this. Performance after load is comparable if not even more responsive.

I've set up a second pull request with that version.